### ISSUE TITLE
Valgrind memory leak in `outdated_index_issue_tracker_t::create_report`

### ISSUE THREAD

**Issue by Tryneus** [rethinkdb/rethinkdb#3590]
This is a tiny memory leak, but it showed up commonly when I ran the shard fuzzer with valgrind in `next`:

```
==22564== 912 bytes in 6 blocks are definitely lost in loss record 17 of 18
==22564==    at 0x4C2B1C7: operator new(unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22564==    by 0x1B12204: outdated_index_issue_tracker_t::create_report(uuid_u const&) (outdated_index.cc:143)
==22564==    by 0x1B3D9FB: do_create_new_store(std::vector<threadnum_t, std::allocator<threadnum_t> > const&, int, store_args_t, serializer_multiplexer_t*, scoped_array_t<scoped_ptr_t<store_t> >*, store_view_t**) (file_based_svs_by_namespace.cc:89)
==22564==    by 0x1B4213C: void boost::_bi::list6<boost::_bi::value<std::vector<threadnum_t, std::allocator<threadnum_t> > >, boost::arg<1>, boost::_bi::value<store_args_t>, boost::_bi::value<serializer_multiplexer_t*>, boost::_bi::value<scoped_array_t<scoped_ptr_t<store_t> >*>, boost::_bi::value<store_view_t**> >::operator()<void (*)(std::vector<threadnum_t, std::allocator<threadnum_t> > const&, int, store_args_t, serializer_multiplexer_t*, scoped_array_t<scoped_ptr_t<store_t> >*, store_view_t**), boost::_bi::list1<long&> >(boost::_bi::type<void>, void (* const&)(std::vector<threadnum_t, std::allocator<threadnum_t> > const&, int, store_args_t, serializer_multiplexer_t*, scoped_array_t<scoped_ptr_t<store_t> >*, store_view_t**), boost::_bi::list1<long&>&, int) const (bind.hpp:601)
==22564==    by 0x1B41B75: void boost::_bi::bind_t<void, void (*)(std::vector<threadnum_t, std::allocator<threadnum_t> > const&, int, store_args_t, serializer_multiplexer_t*, scoped_array_t<scoped_ptr_t<store_t> >*, store_view_t**), boost::_bi::list6<boost::_bi::value<std::vector<threadnum_t, std::allocator<threadnum_t> > >, boost::arg<1>, boost::_bi::value<store_args_t>, boost::_bi::value<serializer_multiplexer_t*>, boost::_bi::value<scoped_array_t<scoped_ptr_t<store_t> >*>, boost::_bi::value<store_view_t**> > >::operator()<long>(long&) const (bind_template.hpp:38)
==22564==    by 0x1B41358: pmap_runner_one_arg_t<boost::_bi::bind_t<void, void (*)(std::vector<threadnum_t, std::allocator<threadnum_t> > const&, int, store_args_t, serializer_multiplexer_t*, scoped_array_t<scoped_ptr_t<store_t> >*, store_view_t**), boost::_bi::list6<boost::_bi::value<std::vector<threadnum_t, std::allocator<threadnum_t> > >, boost::arg<1>, boost::_bi::value<store_args_t>, boost::_bi::value<serializer_multiplexer_t*>, boost::_bi::value<scoped_array_t<scoped_ptr_t<store_t> >*>, boost::_bi::value<store_view_t**> > >, long>::operator()() (pmap.hpp:20)
==22564==    by 0x1B42B6B: callable_action_instance_t<pmap_runner_one_arg_t<boost::_bi::bind_t<void, void (*)(std::vector<threadnum_t, std::allocator<threadnum_t> > const&, int, store_args_t, serializer_multiplexer_t*, scoped_array_t<scoped_ptr_t<store_t> >*, store_view_t**), boost::_bi::list6<boost::_bi::value<std::vector<threadnum_t, std::allocator<threadnum_t> > >, boost::arg<1>, boost::_bi::value<store_args_t>, boost::_bi::value<serializer_multiplexer_t*>, boost::_bi::value<scoped_array_t<scoped_ptr_t<store_t> >*>, boost::_bi::value<store_view_t**> > >, long> >::run_action() (callable_action.hpp:31)
==22564==    by 0x1590677: callable_action_wrapper_t::run() (runtime_utils.cc:42)
==22564==    by 0x1580392: coro_t::run() (coroutines.cc:196)
```

**Comment by Tryneus** [comment]
Fix is up in review 2495.

**Comment by Tryneus** [comment]
The fix has been approved and merged to `next` in commit 2b82a38.  Will be in release 1.16.
